src/nvim/os/keyboard.c

```diff
@@ -0,0 +1,285 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check
+// it. PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <uv.h>
+
+#include "nvim/api/private/helpers.h"
+#include "nvim/api/private/defs.h"
+#include "nvim/lua/executor.h"
+#include "nvim/os/keyboard.h"
+#include "nvim/vim.h"
+#include "nvim/version.h"
+
+# if defined(__linux__)
```

Comment:
It is the inside of `#if` which should be indented, not the other way around. Wondering where have you seen code like this, this is a very fresh idea of formatting preprocessor directives in C for me.

Reply:
I did just `git grep '# if'` a huge output came out, so I thought this would be the new way, as some code is of vim, the new added code would be of neovim which is using this

Reply:
for instance see `fileio.c`

Reply:
`fileio.c` had an `#if` inside an `#if`. (Note past tense, it is inherited file.) Pretty sure that if inside were any preprocessor directives then they would be indented further. You may see lots of preprocessor directives which are indented more or less fine in `vim.h`, even though this is an inherited file and still has some flaws. And most extreme examples of nesting in `func_attr.h`: file contains basically nothing, but macros, and it is a new file, not inherited.

The key point is that `#if…` increase indentation like in any other code with regular `if()`, `#else…` is on the same level as corresponding `#if` and `#endif` is both at the same level and makes further directives appear same indented. When nesting preprocessor directives increased indentation applies *only* to lines starting with `#`, not to the lines which are still part of the preprocessor directives, but appear below (i.e. those which are continuation, denoted by continuation character (backslash) on the previous line) and not to the other code.

Reply:
I got this